utils/swift_build_support/swift_build_support/products/swifttsc.py

```diff
@@ -0,0 +1,59 @@
+# swift_build_support/products/swiftpm.py -----------------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+
+from . import product
+from .. import shell
+
+class SwiftTSC(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        return "swift-tools-support-core"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return True
```

Comment:
I don't think that this should be `True` but rather what the user requested (`self.args.build_swiftsc or self.args.build_swift_package_manager`).

Reply:
That's what I thought initially but it seems like we just return True in every product 🤷‍♂️I think conditionality is handled by https://github.com/apple/swift/pull/29723/files#diff-23cfac4b9c8ced6111824fa8012556e2R812

Maybe @benlangmuir knows the purpose of this.

Reply:
I think it was added to be parallel to the should_test and should_install, but you're right - at the time it was introduced it always returned `True` and the real logic was elsewhere.  Now it looks like tensorflow API and playground support are checking the arguments here.  It would be good to make it consistent :-\